packages/domain/lib/di/injector_domain.dart

```diff
@@ -51,8 +52,13 @@ void _initModuleUseCase() {
       GetIt.instance.get<PreferencesLocalRepository>(),
     ),
   );
-  GetIt.instance.registerFactory<AnalyticsUseCase>(
-    () => AnalyticsUseCase(
+  GetIt.instance.registerFactory<AnalyticsButtonUseCase>(
+    () => AnalyticsButtonUseCase(
+      GetIt.instance.get<AnalyticsService>(),
+    ),
+  );
+  GetIt.instance.registerFactory<AnalyticsPageUseCase>(
+    () => AnalyticsPageUseCase(
```

Comment:
naming - usecase is an action, so it should contain the verb in the name
AnalyticsPageUseCase/AnalyticsButtonUseCase is a noun